### PR TITLE
MPI passthru

### PR DESCRIPTION
**Pull request by sheepforce**
Adds the MPI passthru as discussed in #24 as a first step towards a solution of MPI calls.

### CODE REVIEW

File: bagel/default.nix [hunk 1]
```diff
@@ -96,6 +96,8 @@ in stdenv.mkDerivation {
 
   doCheck = true;
 
+  passthru.mpi = { inherit mpi; };
```
Comment (markuskowa) [inline bagel/default.nix:99]: That results in `passthru.mpi.mpi` but we want `passthru.mpi`. It should read `passthru.mpi = mpi` ( or  `passthru = {inherit mpi};` if you want to use the `inherit` )

File: default.nix [hunk 2]
```diff
@@ -70,10 +70,12 @@ let
...
           "--enable-mpi"
-          "MPICC=${self_.mpi}/bin/mpicc"
-          "MPIFC=${self_.mpi}/bin/mpif90"
-          "MPIF90=${self_.mpi}/bin/mpif90"
+          "MPICC=${self.mpi}/bin/mpicc"
+          "MPIFC=${self.mpi}/bin/mpif90"
+          "MPIF90=${self.mpi}/bin/mpif90"
         ];
+
+        passthru.mpi = { mpi = self.mpi; };
```
Comment (markuskowa) [inline default.nix:78]: For a library it is not really needed. Putting `mpi` in `propagatedBuildInputs` is probably more useful in this case.

File: nixpkgs-opt.nix [hunk 3]
```diff
@@ -27,6 +27,8 @@ self: super:
       "MPIFC=${self.mpi}/bin/mpif90"
       "MPIF90=${self.mpi}/bin/mpif90"
     ];
+
+    propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [self.mpi]
```
Comment (markuskowa) [inline nixpkgs-opt.nix:31]:
```suggestion
    propagatedBuildInputs = oldAttrs.propagatedBuildInputs ++ [ self.mpi ];
```

File: gdma/default.nix [hunk 4]
```diff
@@ -32,6 +32,8 @@ stdenv.mkDerivation rec {
 
     hardeningDisable = [ "format" ];
 
+    passthru = { inherit mpi; };
```
Comment (markuskowa) [inline gdma/default.nix:35]: `gdma` does not use MPI.

Comment (sheepforce) [reply gdma/default.nix]: Sorry, this one slipped in. Must have confused it with GPAW.

**Pull request merged**